### PR TITLE
Reformat slack exception notification

### PR DESCRIPTION
**Pull request by wireframe**
Supercedes this pull request: https://github.com/smartinez87/exception_notification/pull/251

Changes:
* add default options for slack configuration (username, icon)
* use slack attachments API for adding exception metadata
* add support for non-rack exceptions
* use filtered_paramters from request to avoid leaking sensitive
  information

### CODE REVIEW

File: lib/exception_notifier/slack_notifier.rb [hunk 1]
```diff
@@ -1,65 +1,60 @@
...
     end
 
-    def enrich_message_with_backtrace(message, exception)
-      backtrace = clean_backtrace(exception).first(10).join("\n")
-      [message, ['*Backtrace:*', backtrace]].join("\n")
+    # see https://api.slack.com/docs/attachments
+    def attachment_fields(exception, request)
+      backtrace = clean_backtrace(exception).first(10).map { |s| "> #{s}" }.join("\n")
+      fields = [
+        attachment_field('Project', Rails.application.class.parent_name, short: true),
```
Comment (smartinez87) [inline lib/exception_notifier/slack_notifier.rb]: this won't work if it's not a Rails app.

**Comment by smartinez87** [comment]
why change the way the class is structured, and move all the code inside the call method?
Looks like it's doing a lot more than just formatting a message.

Also please look that the build fails with this change

**Comment by adamniedzielski** [comment]
@smartinez87 How can I help with this PR to get it merged?

**Comment by smartinez87** [comment]
@adamniedzielski thanks for jumping in!

I think there needs to be rebased against master, and check the build that was failing.

**Comment by adamniedzielski** [comment]
I reviewed the code and I believe that there are a couple of problems with this PR:

1. it changes the structure of the class
2. it's not backwards compatible, because it doesn't use ```ignore_data_if``` option
3. it's going to break outside of Rails, because ```Hash#slice``` is used

I would rather implement this functionality from scratch.

**Comment by wireframe** [comment]
I'm no longer using this gem, so someone else will have to take up this feature if it is still valuable.

The code in this pull request is herby licensed freely for anyone else to modify/improve/etc.

**Pull request closed without merging**